firefox-webext/background.js

```diff
@@ -25,5 +25,19 @@ function reportIssue (tab) {
   });
 }
 
-chrome.contextMenus.onClicked.addListener(function (tab) { reportIssue(tab) });
-chrome.browserAction.onClicked.addListener(function (tab) { reportIssue(tab) });
+chrome.contextMenus.onClicked.addListener(reportIssue);
+chrome.browserAction.onClicked.addListener(reportIssue);
+
+// Add a custom header when the user is reporting an issue.
+chrome.webRequest.onBeforeSendHeaders.addListener(
+  function(details) {
+    details.requestHeaders.push({
+      name: 'X-Reported-With',
+      value: `${reporterID}`
+    });
+
+    return {requestHeaders: details.requestHeaders};
+  },
+  {urls: ["https://webcompat.com/?open=1&*"]},
```

Comment:
yes explained in https://developer.chrome.com/extensions/match_patterns
btw, I have the feeling there is a typo in MDN `s/url/urls/`, but you have it right yourself.
https://developer.mozilla.org/en-US/Add-ons/WebExtensions/API/WebRequest/onBeforeSendHeaders#details